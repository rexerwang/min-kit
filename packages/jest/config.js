const defineJestConfig = require('@tarojs/test-utils-react/dist/jest.js').default

module.exports = defineJestConfig({
  passWithNoTests: true,
  testEnvironment: 'jsdom',
  testMatch: ['<rootDir>/src/__tests__/*.test.ts?(x)'],
  collectCoverageFrom: ['<rootDir>/src/**/*.ts?(x)'],
  coverageReporters: ['clover', 'json', 'lcov'], //['html', 'text-summary', 'cobertura'],
})
