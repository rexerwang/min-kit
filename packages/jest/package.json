{
  "name": "@min-kit/jest",
  "version": "0.5.11",
  "private": true,
  "description": "jest config & utils for internal",
  "author": "rexerwang",
  "license": "MIT",
  "repository": "rexerwang/min-kit",
  "exports": {
    ".": "./utils.ts",
    "./config": "./config.js"
  },
  "devDependencies": {
    "@happy-dom/jest-environment": "^14.7.1",
    "@tarojs/test-utils-react": "^0.1.1",
    "@testing-library/jest-dom": "^6.4.8",
    "@testing-library/react": "^16.0.1"
  }
}
