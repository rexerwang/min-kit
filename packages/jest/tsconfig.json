{
  "extends": "../../tsconfig.root.json",
  "include": ["utils.ts"]
}
