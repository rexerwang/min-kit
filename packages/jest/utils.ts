import TestUtils from '@tarojs/test-utils-react'

export const createTaroTestUtils = () => {
  const testUtils = new TestUtils()
  return Object.assign(testUtils, { unmount: testUtils.unmout })
}

export const spyOnConsole = (methods = ['error', 'warn', 'info', 'debug']) =>
  methods.map((method: any) => {
    const spy = jest.spyOn(console, method)
    spy.mockImplementation(jest.fn())
    return spy
  })

export * from '@testing-library/react'
