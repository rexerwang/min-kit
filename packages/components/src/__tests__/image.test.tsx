import { createTaroTestUtils, render } from '@min-kit/jest'

describe('<MinImage />', () => {
  const previewImageStub = jest.fn()

  jest.doMock('@tarojs/taro', () => ({
    previewImage: previewImageStub,
  }))

  it('should render <MinImage /> toMatchSnapshot', async () => {
    // @ts-ignore
    const { MinImage } = await import('../image')
    const { asFragment } = render(<MinImage className='image' src='./test.jpg' />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should styling with load status', async () => {
    // @ts-ignore
    const { MinImage } = await import('../image')
    const taro = createTaroTestUtils()
    await taro.mount(MinImage, { props: { className: 'image', src: './test.jpg', preview: true } })

    const el: HTMLElement = await taro.queries.waitForQuerySelector('.image')
    expect(el.classList.contains('min-image--load')).toBeTruthy()

    await taro.act(() => {
      taro.fireEvent.error(el)
    })
    expect(el.classList.contains('min-image--error')).toBeTruthy()

    await taro.act(() => {
      taro.fireEvent.load(el)
    })
    expect(el.classList.contains('min-image--load')).toBeFalsy()
    expect(el.classList.contains('min-image--error')).toBeFalsy()

    taro.unmount()
  })

  it('should previewImage by click after loaded', async () => {
    const onClick = jest.fn()
    const onLoad = jest.fn()

    // @ts-ignore
    const { MinImage } = await import('../image')

    const taro = createTaroTestUtils()
    await taro.mount(MinImage, { props: { className: 'image', src: './test.jpg', preview: true, onClick, onLoad } })
    const el = await taro.queries.waitForQuerySelector('.image')

    await taro.act(() => {
      taro.fireEvent.load(el)
    })
    expect(onLoad).toHaveBeenCalled()

    await taro.act(() => {
      taro.fireEvent.click(el)
    })

    expect(onClick).toHaveBeenCalled()
    expect(previewImageStub).toHaveBeenCalledWith({ urls: ['./test.jpg'] })

    taro.unmount()
  })
})
