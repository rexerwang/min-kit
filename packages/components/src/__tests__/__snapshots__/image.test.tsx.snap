// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MinImage /> should render <MinImage /> toMatchSnapshot 1`] = `
<DocumentFragment>
  <taro-image-core
    class="min-image--load image"
    lazy-load="true"
    src="./test.jpg"
  />
</DocumentFragment>
`;
